pytorch_lightning/loggers/wandb.py

```diff
@@ -126,7 +128,7 @@ def experiment(self) -> Run:
                 os.environ['WANDB_MODE'] = 'dryrun'
             self._experiment = wandb.init(
```

Comment:
Should we handle legacy there ?

Reply:
That's a good question. The new wandb CLI appeared in v0.10.0 (9/11/2020). However the `reinit` logic has actually changed for several months (though it's probably the most robust in the latest versions).

Using the latest version of `wandb` should not be a breaking change since `reinit` was forced to `True` and now it will automatically do what should be expected (ie start a run only when none is existing). Also it's now directly accessible through environment variables if users needed to set it to True for some reasons.